### ISSUE TITLE
display current connected wifi in i3bar

### ISSUE THREAD

**Issue by priestoferis** [powerline/powerline#2050]
I guess this is a feature request. I jump about with my laptop quite a lot, so it's important for me to see what wifi I'm connected to, but I haven't found a segment that can do that. I actually looked into `segments/common/net.py` to see if I can add it myself, but it's a bit over my head without any help.

**Comment by PH111P** [comment]
I implemented something like that some time ago; it needs some polishing before it can be merged upstream though: [net.py](https://github.com/PH111P/powerline/blob/develop/powerline/segments/common/net.py)
In particular, 
* `NetworkManagerSegment` works in case you are using `NetworkManager` (currently requires `nmcli`);
* `wireless` is a general purpose wi-fi segment (requires `iwlib`).

**Comment by priestoferis** [comment]
cool! I will try out your code anyway, thanks!

**Comment by priestoferis** [comment]
I cloned your repo, but the `bindings/i3/powerline-i3.py` is missing from it, so I'm not sure how I could use it, and copying net.py in itself caused an error. 

I also see there is a ton of other changes as well. I was planning to figure out how to add a new argument to the spotify player so in case it would take up too much space it could truncate itself (`i3bar` has this nice feature of having long and short segments, so you can have shortened info, if there are a lot of things on the bar), but as far as I gathered your `develop` branch will soon be merged and it also affects the player, so I should look there if I want to contribute something?

**Comment by PH111P** [comment]
Indeed, there are several other changes, some of which probably won't appear upstream. For your purposes, the following code should work (while not using my fork):
``` python
@requires_segment_info
def wireless(pl, segment_info, device=None, format='{quality:3.0%} at {essid}',
    format_down=None):
    '''Returns the current connection quality.
    :param string device:
        the device to use. Per default this segment will try to be smart.
    :param string format:
        the output format
    :param string format_down:
        if set to any other value than ``None``, it will be shown when no wireless connection is present.
    Highlight groups used: ``wireless:quality`` (gradient), ``wireless:down`` alternatively ``wireless:quality`` (gradient)
    '''
    if not device:
        for interface in os.listdir('/sys/class/net'):
            if interface.startswith('wl'):
                device = interface
                break

    try:
        import iwlib
    except ImportError:
        pl.info("Couldn't load iwlib")
        return None if not format_down else [{
            'contents': format_down.format(quality=0, essid=None, frequency=0),
            'highlight_groups': ['wireless:down', 'wireless:quality', 'quality_gradient'],
            'gradient_level': 100
        }]

    stats = iwlib.get_iwconfig(device)
    stats = {a.lower(): stats[a].decode() if isinstance(stats[a], bytes) else stats[a] for a in stats}

    quality = 0
    essid = ''
    if 'essid' in stats:
        essid = stats['essid']
        if 'stats' in stats and 'quality' in stats['stats']:
            quality = stats['stats']['quality']

    if essid == '' or quality == 0:
        return None if not format_down else [{
            'contents': format_down.format(quality=0, essid=None, **stats),
            'highlight_groups': ['wireless:down', 'wireless:quality', 'quality_gradient'],
            'gradient_level': 100
        }]
    return [{
        'contents': format.format(quality=quality/70, **stats),
        'highlight_groups': ['wireless:quality', 'quality_gradient'],
        'gradient_level': 100 * (70 - quality) / 70
    }]
```
(In essence, remove anything related to `click_values`, `payload`, or `auto shrink` and you should be good to go for the mainline repo. (Also, consider replacing `i3bar` with `lemonbar` for an all-in-all better experience; that's BTW also something that happened in my fork, which will not happen upstream))

In general, when contributing, please look refer to this upstream repo, my fork's develop branch won't be merged anytime soon as there are a couple of breaking changes.

**Comment by priestoferis** [comment]
Thanks, I'll try it out. ( I'll also look into these bars. Truth be told, I just started out using a tiling wm with regolith, so I don't really know much about these yet :)   )

**Comment by priestoferis** [comment]
I just realized the `auto shrink` is something that I wanted to do for the spotify segment. Is that some general feature you think you will do upstream? Because then I'll probably just wait :)

**Comment by PH111P** [comment]
That will probably happen, but please don't ask for an ETA ^^.
